src/Mod/Assembly/Gui/ViewProviderAssembly.cpp

```diff
@@ -212,6 +212,32 @@ bool ViewProviderAssembly::canDragObjectToTarget(App::DocumentObject* obj,
     return true;
 }
 
+void ViewProviderAssembly::updateData(const App::Property* prop)
+{
+    auto* obj = static_cast<Assembly::AssemblyObject*>(pcObject);
+    if (prop == &obj->Group) {
+        // Defer the icon update until the event loop is idle.
+        // This ensures the assembly has had a chance to recompute its
+        // connectivity state before we query it.
+        QTimer::singleShot(0, [obj]() {
+            if (!obj || !obj->isAttachedToDocument()) {
```

Comment:
ASAN is killing the test on this line, probably because in some circumstance obj is not an `Assembly::AssemblyObject*`.

Reply:
Ok so the test deletes the document, then obj pointer has become garbage. It tried to access `obj->isAttachedToDocument` and crashes.

Please  check the new version I made. Instead of capturing obj pointer, I capture obj name and doc name and try to get it from the lambda.